lib/uncruft/deprecation_handler.rb

```diff
@@ -73,16 +69,13 @@ def remove_line_number(message)
       message.sub(/(called from( .+ at)? .+):\d+/, '\1')
     end
 
-    def gem_home(message)
-      message.match(%r{(?i:c)alled from( .+ at)? (#{ENV.fetch('GEM_HOME', nil)}/(.+/)*gems)})&.[](2).presence
-    end
-
-    def user_install(message)
-      message.match(%r{(?i:c)alled from( .+ at)? (#{Gem.user_dir}/(.+/)*gems)})&.[](2).presence
+    def gem_path(message)
+      paths = [ENV.fetch('GEM_HOME', nil), Bundler.home.to_s, Gem.user_dir].compact
```

Comment:
I think we can safely assume `bundler` is defined, right?

It looks like we're assuming `Rails` is defined.